net/respondd/src/respondd.c

```diff
@@ -320,27 +384,99 @@ static struct json_object * handle_request(char *request, bool *compress) {
 	}
 }
 
-
-static void serve(int sock) {
-	char input[256];
-	const char *output = NULL;
-	ssize_t input_bytes, output_bytes;
+// Wait for an incomming request and schedule it.
+//
+// 1a. If the schedule is empty, we wait infinite time.
+// 1b. If we have scheduled requests, we only wait for incomming requests
+//     until we reach the scheduling deadline.
+// 2a. If the incomming request was sent to a multicast destination IPv6,
+//     choose a random delay between 0 and max_multicast_delay milliseconds.
+// 2b. If the incomming request was sent to a unicast destination, delay
+//     is zero.
+static void accept_request(struct request_schedule *schedule, int sock,
+                           uint64_t max_multicast_delay) {
+	char input[REQUEST_MAXLEN];
+	ssize_t input_bytes;
 	struct sockaddr_in6 addr;
 	socklen_t addrlen = sizeof(addr);
-	bool compress;
+	char control[256];
+	struct in6_addr destaddr;
+	struct cmsghdr *cmsg;
+
+	int64_t timeout = schedule_idle_time(schedule);
+	if (timeout < 0)
+		return;
+
+	// set timeout to the socket
+	struct timeval t;
+	t.tv_sec = ((uint64_t) timeout) / 1000;
+	t.tv_usec = (((uint64_t) timeout) % 1000) * 1000;
+
+	if (setsockopt (sock, SOL_SOCKET, SO_RCVTIMEO, (char *)&t, sizeof(t)) < 0)
```

Comment:
The `(char *)` cast is not required, setsockopt takes a `void *`. Also, remove the space after the function name to follow the coding style of the rest of the file.